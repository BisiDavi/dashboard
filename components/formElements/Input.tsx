import { useState, ChangeEvent } from "react";
import { TextField, makeStyles, InputAdornment } from "@material-ui/core";
import { FormikErrors, FormikTouched } from "formik";
import { BsEyeSlash, BsFillEyeFill } from "react-icons/bs";
import { formValues } from "../../types";

const useStyles = makeStyles((theme) => ({
    input: {
        margin: theme.spacing(2, 0),
        width: "100%",
    },
}));

type inputValueType = {
    email: string;
    password: string;
};

interface InputProps {
    input: {
        type: string;
        name: string;
        label: string;
        placeholder: string;
    };
    handleChange?: (e: string | ChangeEvent<any>) => void;
    handleBlur?: (e: string | ChangeEvent<any>) => void;
    values?: unknown;
    errors?: FormikErrors<unknown>;
    touched?: FormikTouched<inputValueType> | any;
    formValues?: { selectedField: formValues; selectedIndex: number };
}

export default function Input({
    input,
    handleChange,
    handleBlur,
    values,
    errors,
    touched,
    formValues,
}: InputProps) {
    const [showPassword, setShowPassword] = useState(false);

    function displayPassword() {
        setShowPassword(!showPassword);
    }
    const isPasswordInput = input.type === "password";
    const classes = useStyles();
    return (
        <>
            <TextField
                variant="outlined"
                type={input.type}
                data-testid="text-input"
                fullWidth
                name={input.name}
                //value={
                //    formValues
                //        ? formValues.selectedField[input.name]
                //        : values[input.name]
                //}
                placeholder={input.placeholder}
                defaultValue={
                    formValues && formValues.selectedField[input.name]
                }
                className={classes.input}
                label={input.label}
                onChange={handleChange}
                onBlur={handleBlur}
                helperText={
                    errors &&
                    errors[values[input.name]] &&
                    touched[values[input.name]]
                }
            />
            {isPasswordInput && (
                <InputAdornment position="end" variant="outlined" />
            )}
        </>
    );
}=
