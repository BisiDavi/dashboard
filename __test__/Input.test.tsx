/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import Input from "@components/formElements/Input";

describe("Test the Input component", () => {
    const inputContent = {
        name: "email",
        type: "email",
        label: "User Email",
        placeholder: "johndoe@gmail.com",
    };
    const values = {
        email: "",
    };

    it("ensure valid input", () => {
        expect(render(<Input values={values} input={inputContent} />))
            .toBeTruthy;
    });

    it("Input component snapshot testing", () => {
        expect(
            render(<Input values={values} input={inputContent} />),
        ).toMatchSnapshot();
    });
});
