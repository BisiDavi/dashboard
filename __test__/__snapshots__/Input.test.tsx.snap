// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Input component Input component snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
        data-testid="text-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          User Email
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="email"
            placeholder="johndoe@gmail.com"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-9"
            >
              <span>
                User Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
      data-testid="text-input"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
      >
        User Email
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="email"
          placeholder="johndoe@gmail.com"
          type="email"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-9"
          >
            <span>
              User Email
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
